json-schema/json-schema/src/main/scala/endpoints/algebra/JsonSchemas.scala

```diff
@@ -280,6 +281,12 @@ trait JsonSchemas extends TuplesSchemas with PartialInvariantFunctorSyntax {
       taggedB: Tagged[B]
   ): Tagged[Either[A, B]]
 
+  /** The JSON schema of a coproduct that share the same parent type and thus can be widened to that parent type */
+  def orElseMergeTagged[A: ClassTag, C >: A, B <: C: ClassTag](
+      taggedA: Tagged[A],
+      taggedB: Tagged[B],
+  ): Tagged[C]
```

Comment:
Could we implement it here in terms of `orElse` and `xmap` instead of having it abstract?

Reply:
That's interesting, I didn't think about that, I will try it out.

Reply:
You were right, this works. Cleans up the implementation alot! I left the newly added unit tests to be sure that `orElseMerge` uses `orElse` correctly. Do you agree?

Reply:
Sure!